### PR TITLE
Add single rotare, roll, and zoom actions. Part of #100

### PR DESCRIPTION
**Pull request by matiasdelellis**
Add these actions with their corresponding hotkeys.
* Rotate 90 degrees [Up, DOWN, LEFT, RIGH], with Shift + Arrow keys. 
* Roll camera 90 degress with Alt + Arrow keys.
* Zoom in with z and and Zoom Out with Z (It should be Shift+Z, but it doesn't seem to work) .

There is a small problem in the rotations, but I don't know how to solve it.
If you do several rotations quickly, it cancels the previous rotation, and stops doing it every 90 degrees since it does it where the previous one was canceled.

### CODE REVIEW

File: src/editor/editor.cpp [hunk 1]
```diff
@@ -672,6 +672,56 @@ void Editor::init_actions()
         get_canvas().animate_to_cam_quat(ry * q);
     });
 
+    connect_action(ActionID::VIEW_ROTATE_UP, [this](auto &a) {
```
Comment (carrotIndustries) [inline src/editor/editor.cpp:675]: You can remove quite a lot of code duplication by creating a member function, connect that one and look at `a` to find out which action got triggered.

**Comment by carrotIndustries** [comment]
> If you do several rotations quickly, it cancels the previous rotation, and stops doing it every 90 degrees since it does it where the previous one was canceled.

For that we'll have to add an new method to the Canvas that does relative animations. See the handle_spnav method for an example.

File: src/editor/editor.cpp (hunk 1 above)
Comment (matiasdelellis) [reply src/editor/editor.cpp:675]: Something like this?
```
   //src/editor/editor.cpp
   connect_action(ActionID::VIEW_ROTATE_UP, [this](auto &a) {
        get_canvas().rotate_cam_quat(90.f, glm::vec3(-1, 0, 0));
    });
```

```
// /src/canvas/canvas.cpp
void Canvas::rotate_cam_quat(float angle, const glm::vec3 &rotation)
{
    auto r =  glm::angleAxis(glm::radians(angle), glm::rotate(m_cam_quat, rotation));
    animate_to_cam_quat(r * m_cam_quat);
}
```

Comment (carrotIndustries) [reply src/editor/editor.cpp:675]: No, like https://github.com/dune3d/dune3d/blob/main/src/editor/editor.cpp#L831 and https://github.com/dune3d/dune3d/blob/4ccdfb94288518af00070deab11152ac95c7a42d/src/editor/editor.cpp#L617

Comment (matiasdelellis) [reply src/editor/editor.cpp]: Done.. I would improve some things... ie. I didn't understand why I can't use switch on action, but I see that in the rest of the code they do something similar. 🤔

**Comment by matiasdelellis** [comment]
> > If you do several rotations quickly, it cancels the previous rotation, and stops doing it every 90 degrees since it does it where the previous one was canceled.
> 
> For that we'll have to add an new method to the Canvas that does relative animations. See the handle_spnav method for an example.

Hi @carrotIndustries 
Thanks for the review! 😉 
This seems a little more difficult for me to resolve soon, but I will try to do it.

**Pull request closed without merging**